gcloud-java-compute/src/test/java/com/google/gcloud/compute/DiskInfoTest.java

```diff
@@ -0,0 +1,248 @@
+      StandardDiskConfiguration.builder()
+          .sizeGb(SIZE_GB)
+          .diskType(TYPE)
+          .build();
+  private static final SnapshotDiskConfiguration SNAPSHOT_DISK_CONFIGURATION =
+      SnapshotDiskConfiguration.builder(SNAPSHOT)
+          .sizeGb(SIZE_GB)
+          .diskType(TYPE)
+          .sourceSnapshotId(SNAPSHOT_ID)
+          .build();
+  private static final ImageDiskConfiguration IMAGE_DISK_CONFIGURATION =
+      ImageDiskConfiguration.builder(IMAGE)
+          .sizeGb(SIZE_GB)
+          .diskType(TYPE)
+          .sourceImageId(IMAGE_ID)
+          .build();
+  private static final DiskInfo DISK_INFO = DiskInfo.builder(DISK_ID, DISK_CONFIGURATION)
+      .id(ID)
+      .creationTimestamp(CREATION_TIMESTAMP)
+      .creationStatus(CREATION_STATUS)
+      .description(DESCRIPTION)
+      .licenses(LICENSES)
+      .attachedInstances(ATTACHED_INSTANCES)
+      .build();
+  private static final DiskInfo SNAPSHOT_DISK_INFO =
+      DiskInfo.builder(DISK_ID, SNAPSHOT_DISK_CONFIGURATION)
+          .id(ID)
+          .creationTimestamp(CREATION_TIMESTAMP)
+          .creationStatus(CREATION_STATUS)
+          .description(DESCRIPTION)
+          .licenses(LICENSES)
+          .attachedInstances(ATTACHED_INSTANCES)
+          .lastAttachTimestamp(LAST_ATTACH_TIMESTAMP)
+          .lastDetachTimestamp(LAST_DETACH_TIMESTAMP)
+          .build();
+  private static final DiskInfo IMAGE_DISK_INFO =
+      DiskInfo.builder(DISK_ID, IMAGE_DISK_CONFIGURATION)
+          .id(ID)
+          .creationTimestamp(CREATION_TIMESTAMP)
+          .creationStatus(CREATION_STATUS)
+          .description(DESCRIPTION)
+          .licenses(LICENSES)
+          .attachedInstances(ATTACHED_INSTANCES)
+          .lastAttachTimestamp(LAST_ATTACH_TIMESTAMP)
+          .lastDetachTimestamp(LAST_DETACH_TIMESTAMP)
+          .build();
+
+  @Test
+  public void testToBuilder() {
+    compareDiskInfo(DISK_INFO, DISK_INFO.toBuilder().build());
+    compareDiskInfo(IMAGE_DISK_INFO, IMAGE_DISK_INFO.toBuilder().build());
+    compareDiskInfo(SNAPSHOT_DISK_INFO, SNAPSHOT_DISK_INFO.toBuilder().build());
+    DiskInfo diskInfo = DISK_INFO.toBuilder().description("newDescription").build();
+    assertEquals("newDescription", diskInfo.description());
+    diskInfo = diskInfo.toBuilder().description("description").build();
+    compareDiskInfo(DISK_INFO, diskInfo);
+  }
+
+  @Test
+  public void testToBuilderIncomplete() {
+    DiskInfo diskInfo = DiskInfo.of(DISK_ID, DISK_CONFIGURATION);
+    assertEquals(diskInfo, diskInfo.toBuilder().build());
+    diskInfo = DiskInfo.of(DISK_ID, SNAPSHOT_DISK_CONFIGURATION);
+    assertEquals(diskInfo, diskInfo.toBuilder().build());
+    diskInfo = DiskInfo.of(DISK_ID, IMAGE_DISK_CONFIGURATION);
+    assertEquals(diskInfo, diskInfo.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(ID, DISK_INFO.id());
+    assertEquals(DISK_CONFIGURATION, DISK_INFO.configuration());
```

Comment:
Do you think it's worth checking diskId, lastAttachTimestamp, and lastDetachTimestamp here as well?